### PR TITLE
 fix bug when cardinality is nil

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
it is fix for issue https://github.com/siddontang/go-mysql/issues/173

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
@holys PTAL

File: schema/schema.go [hunk 1]
```diff
@@ -308,7 +308,12 @@ func (ta *Table) fetchIndexesViaSqlDB(conn *sql.DB) error {
 			currentName = indexName
 		}
 
-		currentIndex.AddColumn(colName, cardinality)
+		c, ok := cardinality.(int)
```
Comment (siddontang) [inline schema/schema.go]: can we meet other types (not just int) here?

**Comment by NearTan** [comment]
Some suggestions:
- before push commit, your can use `make test` on cli to run local unit test 
- rebase upstream before create a new pull request could clean your commits list

**Comment by WangXiangUSTC** [comment]
@NearTan Ok

File: schema/schema.go [hunk 2]
```diff
@@ -307,10 +307,10 @@ func (ta *Table) fetchIndexesViaSqlDB(conn *sql.DB) error {
 			currentIndex = ta.AddIndex(indexName)
 			currentName = indexName
 		}
-
-		c, ok := cardinality.(uint64)
-		if ok {
-			currentIndex.AddColumn(colName, c)
+		
+		switch cardinality := cardinality.(type) {
+		case int, int8, int16, int32, int64, uint8, uint16, uint32, uint64:
```
Comment (siddontang) [inline schema/schema.go]: miss uint type

**Comment by holys** [comment]
LGTM.
PTAL again @siddontang

File: schema/schema.go [hunk 3]
```diff
@@ -307,8 +307,27 @@ func (ta *Table) fetchIndexesViaSqlDB(conn *sql.DB) error {
...
+			currentIndex.AddColumn(colName, uint64(c))
+		case int8:
+			currentIndex.AddColumn(colName, uint64(c))
+		case int16:
+			currentIndex.AddColumn(colName, uint64(c))
+		case int32:
+			currentIndex.AddColumn(colName, uint64(c))
+		case int64:
+			currentIndex.AddColumn(colName, uint64(c))
+		case uint8:
```
Comment (siddontang) [inline schema/schema.go]: miss uint type

File: schema/schema.go [hunk 4]
```diff
@@ -307,8 +307,27 @@ func (ta *Table) fetchIndexesViaSqlDB(conn *sql.DB) error {
...
+			currentIndex.AddColumn(colName, uint64(c))
+		case uint16:
+			currentIndex.AddColumn(colName, uint64(c))
+		case uint32:
+			currentIndex.AddColumn(colName, uint64(c))
+		case uint64:
+			currentIndex.AddColumn(colName, uint64(c))
```
Comment (GregoryIan) [inline schema/schema.go]: add default branch

**Comment by siddontang** [comment]
LGTM

**Pull request merged**